### PR TITLE
Block 6

### PR DESCRIPTION
**Pull request by YaroslavGumeniuk**
### All pull requests:
* [ ] Have you considered security (e.g., XSS)?
* [ ] Have you considered desktop, mobile, and tablet form-factors?
* [ ] Have you considered [accessibility best practices](https://www.w3.org/WAI/standards-guidelines/wcag/)?

### Code
* [ ] Are the commits squashed into one commit?
* [ ] Is the [branch name and commit message](CONTRIBUTING.md#conventions-for-branch-names-and-commit-messages) following team convention?
* [ ] Is the code linted?
* [ ] Are all open issues, questions, and concerns by reviewers answered, resolved, and reviewed?

### Quality assurance
* [ ] If applicable, are unit tests written and are they passing?
* [ ] If applicable, are integration tests written and are they passing?
* [ ] Have checks linked in your pull request passed?

### Documentation
* [ ] Are there meaningful code comments included?

## Applicable to community contributors:
* [ ] Have you granted SFRA team access to your fork? [Grant access](CONTRIBUTING.md#community-contributors) 

### Documentation
* [ ] If applicable, is the change log updated?
* [ ] If applicable, have any UI text changes been reviewed by Documentation?
* [ ] If applicable, have any UI implementations been reviewed by the UX team?

### Security
* [ ] If applicable, has Security reviewed this code?
* [ ] If applicable, is a 3PP request submitted?

### CODE REVIEW

File: cartridges/sfra_demo/cartridges/sfra_demo/cartridge/controllers/AppendExample2.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+var server = require('server');
+var x = require('*/cartridge/controllers/AppendExample');
+
+server.extend(x);
```
Comment (mchervonik) [inline cartridges/sfra_demo/cartridges/sfra_demo/cartridge/controllers/AppendExample2.js]: 'x' - is it a variable name that was in the provided cartridge? Please, avoid using meaningless variable names

File: cartridges/sfra_demo/cartridges/sfra_demo/cartridge/controllers/DefaultWorkaroundForRemovingHomeOnURL.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+var server = require('server');
+server.extend(module.superModule);
+
+var cache = require('*/cartridge/scripts/middleware/cache');
+var pageMetaData = require('*/cartridge/scripts/middleware/pageMetaData');
+
+server.replace('Start', cache.applyDefaultCache, function (req, res, next) {
+    var URLUtils = require('dw/web/URLUtils');
```
Comment (mchervonik) [inline cartridges/sfra_demo/cartridges/sfra_demo/cartridge/controllers/DefaultWorkaroundForRemovingHomeOnURL.js]: Why you need URLUtils here?

File: cartridges/sfra_demo/cartridges/sfra_demo/cartridge/controllers/Error.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+var server = require('server');
+server.extend(module.superModule);
+
+server.append('Start', function (req, res, next) {
+
+    var viewData = res.getViewData();
+    viewData.demo1 = {
+        testVar: 'Demo1 cartridge appending to error page'
+    };
+    res.setViewData(viewData);
```
Comment (mchervonik) [inline cartridges/sfra_demo/cartridges/sfra_demo/cartridge/controllers/Error.js]: No need to use getViewData. You can just simply put
res.setViewData({
    demo1: {
        testVar: '....'
     }
});

**Review by mchervonik — CHANGES_REQUESTED** [review]

File: cartridges/sfra_demo/cartridges/sfra_demo/cartridge/controllers/AppendExample2.js (hunk 1 above)
Comment (YaroslavGumeniuk) [reply cartridges/sfra_demo/cartridges/sfra_demo/cartridge/controllers/AppendExample2.js]: Variable renamed.
It was named by author of this cartridge.  I didn't see it.

File: cartridges/sfra_demo/cartridges/sfra_demo/cartridge/controllers/DefaultWorkaroundForRemovingHomeOnURL.js (hunk 2 above)
Comment (YaroslavGumeniuk) [reply cartridges/sfra_demo/cartridges/sfra_demo/cartridge/controllers/DefaultWorkaroundForRemovingHomeOnURL.js]: Removed.

File: cartridges/sfra_demo/cartridges/sfra_demo/cartridge/controllers/Error.js (hunk 3 above)
Comment (YaroslavGumeniuk) [reply cartridges/sfra_demo/cartridges/sfra_demo/cartridge/controllers/Error.js]: Сorrected

**Review by mchervonik — APPROVED** [review]

**Pull request merged**